src/resolve/resolved-dns-scope.c

```diff
@@ -754,7 +755,7 @@ DnsScopeMatch dns_scope_good_domain(
 
                 if ((s->family == AF_INET && dns_name_endswith(domain, "in-addr.arpa") > 0) ||
                     (s->family == AF_INET6 && dns_name_endswith(domain, "ip6.arpa") > 0))
-                        return DNS_SCOPE_MAYBE;
+                        return dns_scope_is_default_route(s) ? DNS_SCOPE_LAST_RESORT : DNS_SCOPE_NO;
```

Comment:
default routes is a concept we only have for classic DNS. it does not exist for llmnr nor mdns.

dns_scope_is_default_route() hence returns false always here, because this is for the mdns scope.

Reply:
Oh indeed. Hm that's a shame.

I'll just drop this bit for now and strike #14735 from the issues, will maybe consider that later.